localparticipant.go

```diff
@@ -674,3 +678,176 @@ func (p *LocalParticipant) cleanup() {
 	p.rpcPendingAcks = &sync.Map{}
 	p.rpcPendingResponses = &sync.Map{}
 }
+
+func (p *LocalParticipant) StreamText(options StreamTextOptions) (*TextStreamWriter, error) {
+	if options.StreamId == nil {
+		streamId := uuid.New().String()
+		options.StreamId = &streamId
+	}
+
+	if options.Attributes == nil {
+		options.Attributes = make(map[string]string)
+	}
+
+	info := TextStreamInfo{
+		baseStreamInfo: &baseStreamInfo{
+			Id:         *options.StreamId,
+			MimeType:   "text/plain",
+			Topic:      options.Topic,
+			Timestamp:  time.Now().UnixMilli(),
+			Size:       options.TotalSize,
+			Attributes: options.Attributes,
+		},
+	}
+
+	header := &livekit.DataStream_Header{
+		StreamId:    info.Id,
+		MimeType:    info.MimeType,
+		Topic:       info.Topic,
+		Timestamp:   info.Timestamp,
+		TotalLength: info.Size,
+		ContentHeader: &livekit.DataStream_Header_TextHeader{
+			TextHeader: &livekit.DataStream_TextHeader{
+				OperationType: livekit.DataStream_CREATE,
+			},
+		},
+	}
+	if options.ReplyToStreamId != nil {
+		if textHeader, ok := header.ContentHeader.(*livekit.DataStream_Header_TextHeader); ok {
+			textHeader.TextHeader.ReplyToStreamId = *options.ReplyToStreamId
+		}
+	}
+
+	writer, err := newTextStreamWriter(info, header, p.engine, options.DestinationIdentities, options.OnProgress)
+	if err != nil {
+		return nil, err
+	}
+	p.engine.OnClose(func() {
+		err := writer.Close()
```

Comment:
A couple of bits here (assuming I am reading the code right)
1. The write goroutine below seems to keep going even if this closed.
2. On close, the streams is trying to write end chunk. Two things there
     a. Does it make sense to write end chunk if only partial was written?
     b. Looks like close could write an end chunk and then the write can continue to write more?

Reply:
Same concerns also for `SendFile` paths below.

Reply:
> Does it make sense to write end chunk if only partial was written?

I think so, end chunk is essential for the other user to close the reader 

> Looks like close could write an end chunk and then the write can continue to write more?

Good catch, let me handle this.